commons/src/main/kotlin/com/simplemobiletools/commons/dialogs/PermissionRequiredDialog.kt

```diff
@@ -4,20 +4,24 @@ import android.app.Activity
 import androidx.appcompat.app.AlertDialog
 import com.simplemobiletools.commons.R
 import com.simplemobiletools.commons.extensions.getAlertDialogBuilder
-import com.simplemobiletools.commons.extensions.openNotificationSettings
 import com.simplemobiletools.commons.extensions.setupDialogStuff
 import kotlinx.android.synthetic.main.dialog_message.view.*
 
-class PermissionRequiredDialog(val activity: Activity, textId: Int) {
+class PermissionRequiredDialog(
+    val activity: Activity,
+    textId: Int,
+    private val positiveActionCallback: () -> Unit,
+    private val negativeActionCallback: (() -> Unit)? = null
+) {
     private var dialog: AlertDialog? = null
 
     init {
         val view = activity.layoutInflater.inflate(R.layout.dialog_message, null)
         view.message.text = activity.getString(textId)
 
         activity.getAlertDialogBuilder()
-            .setPositiveButton(R.string.grant_permission) { dialog, which -> activity.openNotificationSettings() }
-            .setNegativeButton(R.string.cancel, null).apply {
+            .setPositiveButton(R.string.grant_permission) { _, _ -> positiveActionCallback.invoke() }
```

Comment:
you can shorten it to positiveActionCallback() if it isnt nullable

Reply:
updated!